test/integration/targets/nxos_ip_interface/tests/cli/sanity.yaml

```diff
@@ -0,0 +1,79 @@
+---
+- debug: msg="START TRANSPORT:CLI nxos_ip_interface sanity test"
+
+- set_fact: testint1="{{ nxos_int1 }}"
+- set_fact: testint2="{{ nxos_int2 }}"
+
+- name: "Put interface {{testint1}} into default state"
+  nxos_config: &intdefault1
+    lines:
+      - "default interface {{ testint1 }}"
+    provider: "{{ cli }}"
+  ignore_errors: yes
+
+- name: "Put interface {{testint2}} into default state"
+  nxos_config: &intdefault2
+    lines:
+      - "default interface {{ testint2 }}"
+    provider: "{{ cli }}"
+  ignore_errors: yes
+
+- name: "Make {{testint1}} a layer3 interface"
+  nxos_interface: &l3int1
+    interface: "{{ testint1 }}"
+    mode: layer3
+    description: 'Configured by Ansible - Layer3'
+    admin_state: 'up'
+    state: present
+    provider: "{{ cli }}"
+
+- name: "Make {{testint2}} a layer3 interface"
+  nxos_interface: &l3int2
+    interface: "{{ testint2 }}"
+    mode: layer3
+    description: 'Configured by Ansible - Layer3'
+    admin_state: 'up'
+    state: present
+    provider: "{{ cli }}"
+
```

Comment:
This will work with the following added here without any checks:

```
- name: Clear interface v4
  nxos_ip_interface:
    interface: "{{ testint1 }}"
    version: v4
    state: absent
    addr: 20.20.20.20
    mask: 24
    provider: "{{ cli }}"

- name: Clear interface v6
  nxos_ip_interface:
    interface: "{{ testint2 }}"
    version: v6
    state: absent
    addr: 'fd56:31f7:e4ad:5585::1'
    mask: 64
    provider: "{{ cli }}"
```

And then the equivalent on the nxos side